### PR TITLE
Use a prebuilt docker image for travis build

### PR DESCRIPTION
**Pull request by jackgerrits**
Docker image contains all dependencies pre-installed and reduces build time from ~40 minutes to ~20 minutes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
+FROM ubuntu:16.04 AS build
```
Comment (rajan-chari) [inline Dockerfile]: Should we switch to ubuntu 14.x?

Comment (jackgerrits) [reply Dockerfile]: We should, working on an update now.

**Comment by rajan-chari** [comment]
Looks good.  Thanks

**Comment by JohnLangford** [comment]
Merged, thanks :-)  The smaller travis build times alone make this very useful :-)

**Pull request merged**